ENH: Include QA/QC of unprocessed data

oesteban opened the pull request:
Specifies the QA/QC of unprocessed data with MRIQC.

Resolves: #347.

cc/ @celprov (to be developed by the two of us)

oesteban committed:
enh: draft QA/QC of unprocessed data

Initiates a foundation for the QA/QC of unprocessed data with MRIQC.

Resolves: #347.

celprov committed:
enh: copy paste exclusion criteria from our paper

oesteban committed:
enh: starting to tame the new contents

oesteban commented on docs/data-management/qaqc-criteria.md:
```diff
@@ -0,0 +1,196 @@
+
+## Anatomical MRI
+
+Given our planned analysis, the T1w image will be used for the spatial alignment with the standard MNI152NLin2009cAsym template.
+In addition, surface reconstructions from the T1w image will guide the co-registration of structural and functional (BOLD) images in fMRIPrep.
+Since the latter preprocessing steps are relatively robust to structural images with mild artifacts, the exclusion criteria for unprocessed T1w images are lenient.
+However, individual T1w images may be excluded without such a decision lead to exclusion of the whole session in which it belongs.
+We annotate subjects with visible artifacts in the T1w images in order to ensure rigorous scrutinizing of spatial normalization and surface reconstruction outputs from fMRIPrep (if both modalities passed the first QC checkpoint with MRIQC).
```
This is good -- I think this part is important to lay out how images will be assessed.

One important thing to bear in mind: we need to be clear on the "exclusion level". Individual T1w or T2w images can be excluded if they are really bad, but that does not lead to excluding the full session. That is in contrast to fMRI tasks and rest, as well as DWI, where an instance of low quality that is excluded means the full session is excluded.

oesteban commented on docs/data-management/qaqc-criteria.md:
```diff
@@ -0,0 +1,196 @@
+We annotate subjects with visible artifacts in the T1w images in order to ensure rigorous scrutinizing of spatial normalization and surface reconstruction outputs from fMRIPrep (if both modalities passed the first QC checkpoint with MRIQC).
+The explanation and the description of the flagging criteria 1 to 7 are the same as their counterpart in the previous section.
+
+## Summary
+
+#. Artifactual structures in the background
+#. Susceptibility distortion artifacts
+#. Aliasing ghosts
+#. Wrap-around that overlaps with the brain
+#. Data formatting issues
+#. Motion-related and Gibbs ringing
+#. Extreme intensity non-uniformity
+#. Eye spillover
```
This section should be easy to read and direct. Needing a summary suggests the opposite: some metadiscourse is necessary to be able to delve into the section. We should avoid this.

oesteban commented on docs/data-management/qaqc-criteria.md:
```diff
@@ -0,0 +1,196 @@
+
+### Motion-related and Gibbs ringing
+
+Large head motion during the acquisition of T1w images often expresses itself with the appearance of concentric ripples throughout the scan.
+In the most subtle cases, motion-related ripples look similar to the fine lines generated by Gibbs ringing.
+The latter emerges as a consequence of the truncation of the Fourier series approximation and appears as multiple fine lines immediately adjacent and parallel to high-contrast interfaces.
+While Gibbs ringing is limited to the adjacency of sharp steps in intensity at tissue interfaces, the ripples caused by motion generally span up to the whole brain, and are primarily visible in the sagittal view of MRIQC’s mosaic views.
```
Overall, this section should not *try to school* ourselves or others about artifacts etc.

Instead, it should state clear exclusion criteria in a very specific way.

Perhaps we should move this part from the `mriqc.md` file here:

```md
- [ ] Visualize the first mosaic (zoomed into brain mask):
    - [ ] Check for signal *ripples* around the frontal/prefrontal cortex typically caused by head motion
    - [ ] Check for signal interference leaked from the eyeballs across the PE direction overlapping with brain tissue.
    - [ ] Check for ghosts within the brain:
        - [ ] Overlapping wrap-around.
        - [ ] Nyquist aliases (typically through PE direction).
        - [ ] Ghosts caused by external elements such as headsets or mirror frames.
    - [ ] Check for other artifacts such as fat shifts or RF spoiling within the brain.
    - [ ] Check for zipper artifacts and other EM interferences.
    - [ ] Check for excessive B<sub>1</sub> field inhomogeneity.
    - [ ] Check for inhomogeneous *salt-and-pepper* noise patterns.
    - [ ] Check for global *salt-and-pepper* noise distribution.
- [ ] Scroll down to the background-enhanced mosaic:
    - [ ] Check for signal *ripples* around the head typically caused by head motion.
    - [ ] Check for signal interference leaked from the eyeballs across the PE direction.
        If present, go back to the brain mosaic to confirm whether they overlap with the brain tissue.
    - [ ] Check for ghosts outside the brain, and evaluate whether they may overlap with brain tissue:
        - [ ] Overlapping wrap-around.
        - [ ] Nyquist aliases (typically through PE direction).
        - [ ] Ghosts caused by external elements such as headsets or mirror frames.
    - [ ] Check for other artifacts such as RF spoiling or objects outside the head.
    - [ ] Check for zipper artifacts and other EM interferences.
    - [ ] Check for global *salt-and-pepper* noise distribution in the background.
    - [ ] Check for anomalous noise distribution in the background.
```

In `mriqc.md` we just keep:

```md
- [ ] Visualize the first mosaic (zoomed into brain mask) and apply the [exclusion criteria](qaqc-criteria.md#anchor-to-section)
- [ ] Scroll down to the background-enhanced mosaic and apply the [exclusion criteria](qaqc-criteria.md#anchor-to-section)
```

While in `qaqc-criteria.md` we have:

```md
## Anatomical

### Brain mosaic:
- [ ] Check for signal *ripples* around the frontal/prefrontal cortex typically caused by head motion.
    Exclude this particular T1w if ripples are clear and globally localized.
    These T1w images could degrade the quality of surface reconstruction. 
- [ ] Check for signal interference leaked from the eyeballs across the PE direction overlapping with brain tissue.
    Exclude this particular T1w if the leaked signal substantially overlaps cortical brain areas.
    These T1w images could degrade the quality of surface reconstruction. 
- [ ] Check for ghosts within the brain:
      - [ ] Overlapping wrap-around.
      - [ ] Nyquist aliases (typically through PE direction).
      - [ ] Ghosts caused by external elements such as headsets or mirror frames.
      
    Exclude this particular T1w if any of these ghosts overlap cortical gray matter.

- [ ] Check for other artifacts such as fat shifts or RF spoiling within the brain.
   Exclude this particular T1w if any of these ghosts overlap cortical gray matter.
- [ ] Check for zipper artifacts and other EM interferences.
    Exclude this particular T1w if any of these ghosts overlap cortical gray matter.
- [ ] Check for excessive B<sub>1</sub> field inhomogeneity.
    Exclude only if it is evident that a coil failure happened.
- [ ] Check for inhomogeneous *salt-and-pepper* noise patterns.
    Generally, do not exclude this T1w image unless the noise pattern destroys cortical gray matter areas.
- [ ] Check for global *salt-and-pepper* noise distribution.
    Generally, do not exclude this T1w image except evident imaging global failure.

### Background-enhanced mosaic
- [ ] Check for signal *ripples* around the head typically caused by head motion.
    Exclude this T1w only if identifying these ripples leads to revising the decision on the brain mosaic.
- [ ] Check for signal interference leaked from the eyeballs across the PE direction.
    Exclude this T1w only if identifying these leakages leads to revising the decision on the brain mosaic.
- [ ] Check for ghosts outside the brain, and evaluate whether they may overlap with brain tissue:
        - [ ] Overlapping wrap-around.
        - [ ] Nyquist aliases (typically through PE direction).
        - [ ] Ghosts caused by external elements such as headsets or mirror frames.
    Exclude this T1w only if identifying these ghosts leads to revising the decision on the brain mosaic.

... continued ...

```

celprov replied:
Yes, I like the separation of having the checklist of exclusion criteria in `qaqc-criteria.md` and in `mriqc.md` have one line saying "apply those exclusion criteria".

Ok I understand your intention of being very brief and precise in exclusion criteria statement. But what if me or the reader doesn't know what this artifact is?
Part of the solution is to refer to our paper, but e.g. I don't know what is `salt-and-pepper` noise distribution and we didn't cover this in the paper.

I also had the idea to add figures to illustrate the exclusion criteria, but I guess this fall under the *schooling* approach. As such that's not what we are going for today. Can you confirm?

oesteban reviewed (commented):
@celprov before you continue working on this remember to `git pull`

I've started working on this and realized that copy&paste was a bad strategy. There's no interest in re-reading the paper, and much less for ourselves, so the content copied is doomed to be beautifully (because the SOPs look beautiful) trashed.

Please look at the inline comments for suggestions.

celprov committed:
fix: apply suggestions to separate exclusion criteria checklist from mriqc procedure
fix: delete copy pasted content from paper

celprov committed:
enh: complete fMRI exclusion criteria taking inspiration from our paper.
enh: separate fMRI exclusion criteria into the ones applicable to all types and the ones specific to each type of fMRI scan.

celprov committed:
enh: add formal QC steps

celprov committed:
enh: leverage MRIQC group report for QC - criteria inspired from Birn 2023 paper

celprov committed:
enh: add exclusion criteria inspired from Etzel 2023 paper

celprov committed:
fix: move exclusion criteria duplicated for each fMRI type under the exclusion criteria for all fMRI type section

celprov committed:
fix: adapt the SOP to the position swap of background and zoom mosaics
enh: add links to less well-known artifacts

celprov committed:
Merge branch 'enh/add-mriqc-exclusion-criteria' of github.com:TheAxonLab/hcph-sops into enh/add-mriqc-exclusion-criteria

celprov committed:
enh: add exclusion criteria inspired from Lepping 2023

celprov committed:
enh: add SNR related criteria

oesteban commented on docs/data-management/mriqc.md:
```diff
@@ -10,10 +10,96 @@
+        │   ├── sub-001_ses-001_acq-original_T2w.html
+        │   ├── sub-001_ses-001_acq-undistorted_T1w.html
+        │   ├── sub-001_ses-001_acq-undistorted_T2w.html
+        │   ├── sub-001_ses-001_task-bht_dir-AP_part-mag_bold.html
+        │   ├── sub-001_ses-001_task-bht_dir-AP_part-phase_bold.html
+        │   ├── sub-001_ses-001_task-qct_dir-AP_part-mag_bold.html
+        │   ├── sub-001_ses-001_task-qct_dir-AP_part-phase_bold.html
+        │   ├── sub-001_ses-001_task-rs_dir-AP_part-mag_bold.html
+        │   ├── sub-001_ses-001_task-rs_dir-AP_part-phase_bold.html
```
```suggestion
        │   ├── sub-001_ses-001_task-bht_dir-AP_bold.html
        │   ├── sub-001_ses-001_task-qct_dir-AP_bold.html
        │   ├── sub-001_ses-001_task-rest_dir-AP_bold.html
```

oesteban commented on docs/data-management/mriqc.md:
````diff
@@ -10,10 +10,96 @@
         "{inputs} {outputs} participant --session-id ${lastsession} -w ${HOME}/tmp/hcph-derivatives/mriqc-23.1.0 --mem 40G"
     ```
+- [ ] Check that *MRIQC* generated all expected derivatives. For each session, you should have the following files:
+    ``` {.shell hl_lines="22-23 40-41 58-59"}
````
I don't think you want to highlight anything
```suggestion
    ``` text
```

oesteban commented on docs/data-management/mriqc.md:
````diff
@@ -10,10 +10,96 @@
         --output ./derivatives/mriqc-23.1.0 \
         "{inputs} {outputs} participant --session-id ${lastsession} -w ${HOME}/tmp/hcph-derivatives/mriqc-23.1.0 --mem 40G"
     ```
+- [ ] Check that *MRIQC* generated all expected derivatives. For each session, you should have the following files:
````
```suggestion
- [ ] Check that *MRIQC* generated all expected individual reports.
    For each session, the following files must be generated:
```

oesteban commented on docs/data-management/mriqc.md:
````diff
@@ -10,10 +10,96 @@
+        │   ├── sub-001_ses-001_task-rs_dir-AP_part-phase_bold.html
+    ```
+
+??? warning "Not all *MRIQC* derivatives were generated"
+
+    - [ ] If some derivatives are missing, it is a sign that *MRIQC* encountered an error. Identify what failed in the "About > Errors" section of the visual report or in the log of the *MRIQC* run.
+    - [ ] If you could find the solution to the problem, re-run *MRIQC* on that particular subject implementing this solution.
+    - [ ] In case you do not understand the error message, search for associated keywords in the [issues of the *MRIQC* github repository](https://github.com/nipreps/mriqc/issues); it is likely someone else experienced the same problem before you and reported it. The solution might be documented in the issue. Don't forget to check closed issues!
+    - [ ] If that did not help, you might find help on [NeuroStars](https://neurostars.org/).
+    - [ ] If no issue has been opened regarding this error and you did not find answer in NeuroStars, then open an issue in the *MRIQC* github repository. Your description of the problem need to be as complete and detailed as possible to help the maintainers identify the problem efficiently.
+
````
If the visual reports are not generated, you cannot check inside them. Let's move that to later (when we have already opened the report), and check here the things you can right away:

```suggestion
    ??? bug "*MRIQC* failed to produce all the expected visual reports"

        Depending on the specific error condition hit by *MRIQC*, some visual reports may not be generated at all.
        
        - [ ] Check for corresponding *crash files* under the `logs/` directory under the output folder.
        - [ ] Address the issue (e.g., insufficient disk space) and re-run *MRIQC*.
            If the issue is unclear, search for similar problems reported previously in the [issues of the *MRIQC* GitHub repository](https://github.com/nipreps/mriqc/issues).
            The solution could be documented in some existing issue opened by a user who experienced the same problem before you and reported it.
            Don't forget to check closed issues!
            If that did not help, you might find help on [NeuroStars](https://neurostars.org/).
        - [ ] Open a new issue in the *MRIQC* GitHub repository if all the above fails.

- [ ] Check that the `group_{T1w,T2w,bold,dwi}.tsv` file contains the image quality metrics (IQMs) for all the expected inputs (one row per individual run at the input).
```

celprov replied:
true ahah XD

oesteban commented on docs/data-management/mriqc.md:
```diff
@@ -10,10 +10,96 @@
+In addition, *MRIQC* is executed prior any further processing step considering our preliminary findings regarding *defacing*<sup>[2]</sup>.
+
+### Assessing anatomical images (T<sub>1</sub>-weighted and T<sub>2</sub>-weighted)
+- [ ] Open each *MRIQC* report on a current Web Browser (*Google Chrome* is preferred).
```
```suggestion
- [ ] Open each *MRIQC* report on a current Web Browser (*Google Chrome* is preferred).
- [ ] Check that the visual report is complete when you open it.

    ??? bug "A visual report is incomplete"

        - [ ] Identify what failed in the "About > Errors" section of the visual report.
        - [ ] Address the issue (e.g., out-of-memory when running a container) and re-run *MRIQC*.
            Proceed as in the previous case by finding documentation on the *MRIQC* repository or NeuroStars.

```

oesteban commented on docs/data-management/mriqc.md:
```diff
@@ -10,10 +10,96 @@
+- [ ] Visualize the first mosaic (background-enhanced mosaic) and apply the [exclusion criteria](qaqc-criteria.md#background-enhanced-mosaic)
+- [ ] Scroll down to the zoom in the brain mask mosaic and apply the [exclusion criteria](qaqc-criteria.md#zoomed-in-brain-mosaic)
+- [ ] Assign a quality rating and indicate artifacts with the *Rating widget*.
+    To assign a quality rating, follow the [QA/QC criteria]() stated below.
```
missing link, also the criteria are on a different document (not below)

oesteban commented on docs/data-management/mriqc.md:
```diff
@@ -10,10 +10,96 @@
+
+### Assessing functional images
+
+!!! danger "Insufficient quality of an fMRI run requires recalling the session"
```
```suggestion
!!! danger "Insufficient quality of an RSfMRI run requires recalling the session"
```

oesteban replied:
as per our discussion today

oesteban commented on docs/data-management/mriqc.md:
```diff
@@ -10,10 +10,96 @@
+
+#### RSfMRI
+- [ ] Open each *MRIQC* report on a current Web Browser (*Google Chrome* is preferred).
+- [ ] Visualize the first mosaic (BOLD average) and apply the [exclusion criteria](qaqc-criteria.md#bold-average)
```
The new reports show the standard deviation first and then the carpet plot. Please check them :)

oesteban commented on docs/data-management/mriqc.md:
```diff
@@ -10,10 +10,96 @@
+- [ ] Scroll down to the background noise mosaic and apply the [exclusion criteria](qaqc-criteria.md#background-noise-mosaic)
+- [ ] Scroll down to the fMRI summary plot and apply the [exclusion criteria](qaqc-criteria.md#fmri-summary-plot) as well as [the exclusion criteria specific to the BHT](qaqc-criteria.md#resting-state).
+
+#### QCT
+- [ ] Open each *MRIQC* report on a current Web Browser (*Google Chrome* is preferred).
+- [ ] Visualize the first mosaic (BOLD average) and apply the [exclusion criteria](qaqc-criteria.md#bold-average)
+- [ ] Scroll down to the standard-deviation (std) mosaic and apply the [exclusion criteria](qaqc-criteria.md#standard-deviation-mosaic)
+- [ ] Scroll down to the background noise mosaic and apply the [exclusion criteria](qaqc-criteria.md#background-noise-mosaic)
+- [ ] Scroll down to the fMRI summary plot and apply the [exclusion criteria](qaqc-criteria.md#fmri-summary-plot) as well as [the exclusion criteria specific to the BHT](qaqc-criteria.md#quality-control-task) 
```
I think we should look at the QCT the first:
- to train our eye
- to identify sessions where fMRI may have problems that went beyond the qct

oesteban commented on docs/data-management/mriqc.md:
```diff
@@ -10,10 +10,96 @@
+- [ ] Scroll down to the background noise mosaic and apply the [exclusion criteria](qaqc-criteria.md#background-noise-mosaic)
+- [ ] Scroll down to the fMRI summary plot and apply the [exclusion criteria](qaqc-criteria.md#fmri-summary-plot) as well as [the exclusion criteria specific to the BHT](qaqc-criteria.md#quality-control-task) 
+
+#### BHT
```
I think we should evaluate the bht the second, for similar reasons.

oesteban commented on docs/data-management/mriqc.md:
```diff
@@ -10,10 +10,96 @@
+
+### Assessing diffusion images
+
+
```
```suggestion
```

oesteban commented on docs/data-management/mriqc.md:
```diff
@@ -10,10 +10,96 @@
+### Assessing diffusion images
+
+
+!!! danger "Insufficient quality of a dMRI run requires recalling the session"
```
Not sure we want to do this, I think we can drop dMRI sessions without collecting a full session again.

celprov replied:
I was thinking the point was to be able to compute FC and SC for each session.
As such, why do we recollect a session in case of RSfMRI exclusion but not dMRI?

oesteban replied:
Statistical power

celprov replied:
So we expect effect sizes to be smaller in FC than in SC?

celprov replied:
Let's move this to an issue, to be able to close this PR.

oesteban commented on docs/data-management/post-session.md:
```diff
@@ -423,6 +423,29 @@ As new sessions are collected, the corresponding BIDS structures MUST be saved w
 
     !!! danger "Always double-check that data in the annex are uploaded to the RIA store"
 
+### Formal QC
+The BIDS validator automatize the verification that the files are in the right format, that the filenames follow [the BIDS specifications](https://bids-specification.readthedocs.io/en/stable/), that each session contains the same modalities and the consistency of parameters across sessions.
+
+- [ ] Open the [BIDS validator](https://bids-standard.github.io/bids-validator/) in a Web browser.
+
+- [ ] Click on <span class='keypress'>Browse</span> 
+- [ ] A pop up window will appear
+    
+    > **bids-standard.github.io**
+    > 
+    > Are you sure you want to upload all files from “Papers”? Only do this if you trust the site.
+    
+    Click <span class='keypress'>Upload</span>. You do not have to worry, no data will be uploaded to the Web interface, only the folder structure.
+
+- [ ] Verify that there is no unexpected error or warning. If it is not the case, identify the problem and repeat the BIDS conversion of the affected files.
```
I'd rather use the bids validator packed, e.g., within the fmriprep container.

```suggestion
### Formal QC

- [ ] Run the *BIDS Validator* to check the *formal* quality of the dataset (filenames, homogeneity of modalities and parameters across sessions, etc.)
    ``` shell
    docker run ...
    ```
```

- [ ] Only the following **ERRORS** are expected:
    <!-- list errors here, e.g., the eye tracker -->
- [ ] Only the following **WARNINGS** are acceptable:
    <!-- list warnings here, e.g., the different pe directions -->

oesteban commented on docs/data-management/qaqc-criteria.md:
```diff
@@ -0,0 +1,175 @@
+    Generally, do not exclude this T1w image except evident imaging global failure.
+- [ ] Check for low SNR characterized by a grainy picture. Generally, do not exclude this T1w image unless the noise pattern destroys cortical gray matter areas.
+
+... continued ...
```
please remove

oesteban commented on docs/data-management/qaqc-criteria.md:
```diff
@@ -0,0 +1,175 @@
+... continued ...
+
+### Group report
+- If they are not excluded yet, re-inspect the individual visual report of the three scans with the lowest SNR and double-check that the data does not fall into one of the [exclusion criteria](#functional-mri).
```
```suggestion
- [ ] Check again the individual visual report of runs with outlying-low SNR, ensuring they do not fall into one of the [exclusion criteria](#functional-mri).
```

oesteban reviewed (commented):
Shaping up!

celprov committed:
Apply suggestions from code review

Co-authored-by: Oscar Esteban <code@oscaresteban.es>

celprov committed:
fix: overhaul of the unprocessed QC protocol
fix: very lenient QC protocol for task fMRI
fix: adapt order to new MRIQC reports

celprov commented:
Let's merge this for now and continue working on its development in another PR.
